lib/galaxy/model/__init__.py

```diff
@@ -11569,6 +11569,73 @@ def __repr__(self):
         )
 
 
+class UserCredentials(Base):
+    """
+    Represents a credential associated with a user for a specific service.
+    """
+
+    __tablename__ = "user_credentials"
+
+    id: Mapped[int] = mapped_column(primary_key=True)
+    user_id: Mapped[int] = mapped_column(ForeignKey("galaxy_user.id"), index=True, nullable=False)
+    reference: Mapped[str] = mapped_column(nullable=False)
```

Comment:
We may need to define a unique constraint on multiple fields to avoid duplicate credentials for a given reference.